### PR TITLE
[Arm64] Fix smul[lh] disassembly

### LINKED ISSUE #13834: [Arm64] Incorrect disassembly
```
        9B407D29          smulh   w9, w9, w0
        9B207D29          smull   w9, w9, w0
```

Should be 

```
        9B407D29          smulh   x9, x9, x0
        9B207D29          smull    x9, w9, w0
```

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #13834 

@dotnet/arm64-contrib PTAL

### CODE REVIEW

File: src/jit/emitarm64.cpp [hunk 1]
```diff
@@ -10824,6 +10824,15 @@ void emitter::emitDispIns(
...
                 emitDispReg(encodingZRtoSP(id->idReg2()), size, true);
             }
+            else if ((ins == INS_smull) || (ins == INS_smulh))
+            {
+                // Rd is always 8 bytes
+                emitDispReg(encodingZRtoSP(id->idReg1()), EA_8BYTE, true);
+
+                // Rn, Rm effective size depends on instruction type
+                size = (ins == INS_smulh) ? EA_8BYTE : EA_4BYTE;
+                emitDispReg(encodingZRtoSP(id->idReg2()), size, true);
```
Comment (sdmaclea) [inline src/jit/emitarm64.cpp]: The `encodingZRtoSP` looks wrong.

Comment (sdmaclea) [reply src/jit/emitarm64.cpp]: Done

**Comment by sdmaclea** [comment]
@dotnet/jit-contrib PTAL

**Comment by briansull** [comment]
I would like to wait until PR #14325 is checked in before merging this

**Pull request merged**